### PR TITLE
Replace utcnow (and fix style along the way)

### PR DESCRIPTION
**Pull request by mraspaud**
This PR replaces usages of utcnow in trollmoves.
Ruff wasn't happy about some of the code, so style was modified accordingly.

### CODE REVIEW

File: bin/remove_it.py [hunk 1]
```diff
@@ -231,7 +246,7 @@ def clean_section(pub, section, conf, is_dry_run=True):
             kws[key] = int(info[key])
         except KeyError:
             pass
-    ref_time = datetime.utcnow() - timedelta(**kws)
+    ref_time = dt.datetime.now(dt.UTC) - dt.timedelta(**kws)
```
Comment (pnuu) [inline bin/remove_it.py]: `dt.UTC` is added in 3.11 so it won't work with Python 3.10 which we support. Until we drop 3.10 we should use `dt.timezone.utc` instead.

**Review by pnuu — CHANGES_REQUESTED** [review]
For 3.10 support we need to use `dt.timezone.utc`.

File: bin/remove_it.py (hunk 1 above)
Comment (mraspaud) [reply bin/remove_it.py]: Yep, fixed

**Review by pnuu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**